api/api.py

```diff
@@ -100,33 +101,50 @@ def host_response():
         action = str(request.json['action'])
         target = request.json['target']
         if action == 'blacklist':
-            utils.flush_ipset_list('blacklist-user')
-            for ip in target:
-                with lock:
-                    if (len(ip) >= 7) and (ip not in homenet.user_blacklist) and (ip not in homenet.user_whitelist):
-                        homenet.user_blacklist.append(ip)
-                        utils.add_ip_ipset_blacklist(ip, 'blacklist-user')
-                        log.debug('FG-INFO: IP ' + ip + 'added to user blacklist')
+            if re.search('[a-zA-Z]', target):
```

Comment:
Why do you evaluate a regex on "target" which is a list? This step should be executed on each item of the list instead and I suggest to use a proper regex to ensure that the user is sending strings which are real domains.